### PR TITLE
Add ability for char_map to emit length, plus a few more unit tests t…

### PR DESCRIPTION
**Pull request by apark263**
This adds a configuration option `emit_length` to `char_map` such that an additional output buffer will be created giving the valid length of each character sequence.  This is useful for post-processing a sequence buffer for ctc.  

I also added some tests to do end-to-end provision of a unicode character string as it might be encountered in a manifest file.

### CODE REVIEW

File: test/test_provider_audio.cpp [hunk 1]
```diff
@@ -82,6 +82,79 @@ TEST(provider, audio_classify)
...
+        {
+            if (j < expected_encodings[i].size())
+            {
+                EXPECT_EQ(loaded_transcript[j], expected_encodings[i][j]);
+            }
+            else
+            {
+                EXPECT_EQ(loaded_transcript[j], 0);
+            }
+
```
Comment (pzelazko) [inline test/test_provider_audio.cpp]: unnecessary empty line

File: test/test_provider_audio.cpp [hunk 2]
```diff
@@ -82,6 +82,79 @@ TEST(provider, audio_classify)
     }
 }
 
+
+TEST(provider, transcript_length_check)
+{
+    uint32_t       max_length = 15;
```
Comment (pzelazko) [inline test/test_provider_audio.cpp]: automatic formatter could be run over this file

File: test/test_provider_audio.cpp [hunk 3]
```diff
@@ -82,6 +82,79 @@ TEST(provider, audio_classify)
...
+    nlohmann::json js_transcript    = {
```
Comment (pzelazko) [inline test/test_provider_audio.cpp]: automatic formatter could be run over this file

File: test/test_provider_audio.cpp (hunk 1 above)
Comment (pzelazko) [inline test/test_provider_audio.cpp]: unnecessary empty line

**Comment by pzelazko-intel** [comment]
It's already on master
